src/iconv.jl

```diff
@@ -57,6 +57,12 @@ type StringDecoder{S<:IO} <: IO
     skip::Int
 end
 
+# This is called during GC, just make sure C memory is returned, don't throw errors
```

Comment:
Sure - I've just been following the conventions I learned from digging into Base.
`nothing` is fine, I had started using `Void()` because Jeff had added that.